File: wtf.py
#!/usr/bin/env python

def battleship(v):
    l = 'MISS'
    if v:
        l = 'HIT'
    return '{0} : {1}'.format(l, repr(v))

if __name__ == '__main__':
    # WTF OMIT
    print '\n'.join([ battleship(x) for x in [0, '', (), [], None] ])
    # WTFE OMIT

File: wtf.rb
#!/usr/bin/env ruby

def battleship(v)
  l = 'MISS'
  l = 'HIT' if v
  "#{l} : #{v}"
end

def main
 [0, '', [], nil].each do |v|
    puts battleship(v)
  end
end

main
